### PR TITLE
Button Component - Oy2 19468

### PR DESCRIPTION
**Pull request by jdinh8124**
## Purpose
- Create Button component and add it to the storybook. Storybook created documentation with the types of props it accepts

Resolves:
https://qmacbis.atlassian.net/jira/software/c/projects/OY2/boards/240?modal=detail&selectedIssue=OY2-19468

## Type of Request
- Feature Request

#### Pull Request Creator Checklist

- [x] It is documented what type of request this is.
- [x] There is a detailed description for the purpose of this PR.
- [x] This PR is in line with the intention of this Project

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -1,12 +1,74 @@
...
+    <button
+      {...rest}
+      aria-label={ariaLabel || `${buttonText} button`}
+      className={`usa-button usa-button--${buttonVariationType} ${
+        largeButton ? "usa-button--big" : ""
+      }`}
+    >
+      {shiftIconLeft && icon && <span className="margin-right-1">+</span>}
+      <span>{buttonText}</span>
+      {icon && !shiftIconLeft && <span className="margin-left-1">+</span>}
```
Comment (jphechter) [inline src/components/Button/Button.tsx:71]:
```suggestion
      {shiftIconLeft && icon && <span className="margin-right-1">{icon}</span>}
      <span>{buttonText}</span>
      {icon && !shiftIconLeft && <span className="margin-left-1">{icon}</span>}
```

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -1,12 +1,74 @@
...
+  ariaLabel?: string;
+  largeButton?: boolean;
+}
+
+/**
+ * Button Component
+ * @param {string}  [buttonText]        Renders the text contained in the button.
+ * @param {string}  buttonVariation     Renders the style of the button.
+ * @param {string}  disabled            Determines whether or not a button is enabled.
+ * @param {string}  icon                Handles the icon to render in the button.
```
Comment (jphechter) [inline src/components/Button/Button.tsx:20]: Is this always going to be a string or should this accept an actual image/svg?

**Review by jphechter — CHANGES_REQUESTED** [review]
This looks good for the most part. Wrote a question regarding the icon. Also have one about code coverage.

Are we concerned with getting 100% for the branch logic? If so, it looks like it's just the icon checks.
File                       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
---------------------------|---------|----------|---------|---------|-------------------
 src/components/Button     |     100 |    66.66 |     100 |     100 |
  Button.tsx               |     100 |    66.66 |     100 |     100 | 60,69-71

**Comment by jdinh8124** [comment]
> This looks good for the most part. Wrote a question regarding the icon. Also have one about code coverage.
> 
> Are we concerned with getting 100% for the branch logic? If so, it looks like it's just the icon logic that needs coverage.
> 
> File	% Stmts	% Branch	% Funcs	% Lines	Uncovered Line #s
> src/components/Button	100	66.66	100	100	
> Button.tsx	100	66.66	100	100	60,69-71

Thanks for taking a look, think I'm going to get the Icon component all squared up and then worry about merging the button component afterwards. It seems like the icon will be needed first

File: src/components/Button/Button.tsx (hunk 2 above)
Comment (jdinh8124) [reply src/components/Button/Button.tsx]: Its a stringName now to use the Icon component

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (jdinh8124) [reply src/components/Button/Button.tsx]: I updated it to use this logic and bumped up the branch to a higher percentage, I don't think we're super concerned of getting 100% across the board.

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -1,12 +1,80 @@
...
+    <button
+      disabled={disabled}
+      {...rest}
+      aria-label={ariaLabel || `${buttonText} button`}
+      className={`display-flex usa-button usa-button--${buttonVariationType} ${
+        largeButton ? "usa-button--big" : ""
+      }`}
+    >
+      {shiftIconLeft && iconName && <Icon name={iconName} />}
+      <span className="padding-top-05 padding-x-05">{buttonText}</span>
```
Comment (jphechter) [inline src/components/Button/Button.tsx:72]:
```suggestion
      <span className="padding-x-05">{buttonText}</span>
```
This is causing the text to align slightly off-center.

Screen Shot 2022-08-01 at 4 04 04 PM

**Review by jphechter — APPROVED** [review]

File: src/components/Button/Button.tsx (hunk 3 above)
Comment (jdinh8124) [reply src/components/Button/Button.tsx]: Logic was incorporated to clean this up

**Pull request merged**